src/porepy/numerics/ad/equation_system.py

```diff
@@ -399,57 +399,69 @@ def create_variables(
             non_admissible = requested_type.difference(self.admissible_dof_types)
             raise ValueError(f"Non-admissible DOF types {non_admissible} requested.")
 
-        # Sanity check if variable is defined anywhere.
-        if subdomains is None and interfaces is None:
-            raise ValueError(
-                "Cannot create variable not defined on any subdomain or interface."
-            )
-
         # Container for all grid variables.
         variables = []
 
         # Merge subdomains and interfaces into a single list.
-        assert subdomains is not None or interfaces is not None  # for mypy
-        grids: Sequence[pp.GridLike] = subdomains if subdomains else interfaces  # type: ignore
+        grids: Sequence[pp.GridLike]
+        if subdomains is not None and interfaces is None:
+            grids = subdomains
+        elif subdomains is None and interfaces is not None:
+            grids = interfaces
+        elif subdomains is None and interfaces is None:
+            raise ValueError(
+                "Cannot create variable not defined on any subdomain or interface."
+            )
+        else:
+            # Yura: The current behavior is just to ignore the interfaces, which is
```

Comment:
I agree this looks strange, but what do you mean by "ignore interfaces"?

Reply:
If the user passes both subdomains and interfaces, the line `subdomains if subdomains else interfaces` will silently discard the interfaces.

Reply:
But I think passing both subdomains and interfaces is not allowed. If so, the old line `assert subdomains is not None or interfaces is not None` should probably have been an xor. In your changed implementation, I guess this corresponds to replacing the elif on 411 to else, right?

Reply:
Added the error there.